Integrating Akka.F# Lib

Kavignon opened the pull request:
Added the concurrent lib for Game of Light and Shadows. With Akka.FSharp, the architecture becomes reactive and fault tolerant to issues.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gameoflightandshadows/a-game-of-light-and-shadows/65)

<!-- Reviewable:end -->

Commit:
#31 -  Removed the library and added core (thought this was done already...)
Added the depencies required to used Akka.NET with F# and be able to log with Akka
Empty shells for : global state, logging, protocol messaging and for the managers (battle sequence, command, game)

review-ninja commented:
<a href="https://app.review.ninja/GameOfLightAndShadows/A-game-of-light-and-shadows/pull/65" target="_blank"><img src="https://app.review.ninja/66727011/pull/65/badge" alt="ReviewNinja"/>

Kavignon commented:
Reviewed 25 of 25 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/gameoflightandshadows/a-game-of-light-and-shadows/65)*
<!-- Sent from Reviewable.io -->

Commit:
Fixed build issues in AppVeyor

Commit:
Modify the .sln file to represent the actual state of the solution.

Kavignon commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/gameoflightandshadows/a-game-of-light-and-shadows/65)*
<!-- Sent from Reviewable.io -->

Commit:
#57 - Standardize module naming accros the whole project
added empty shells of the remaining akka actors (managers)

Commit:
#56 - Added the shell of message required for the Akka agents. Through time, the message arguments will be updated, some new messages could be added/removed, depending on the state of the project.

Commit:
#58 - Create the global game state which currently has the menu, the battle sequence, the weapon and item store.

Commit:
#58 - Added the game board in the battle sequence and the storyline in the global state

Commit:
Modifications on the .sln file.

Commit:
#5 - Started work for handling the global game state

Commit:
Implemented a basic implementation to update the state when a changes occurs
Shell of the DeadMessageServer will have to subscribe to the EventStream and look for dead messages

Commit:
Implemented shell for the command manager. Lacking way to "tell" the game manager that a character did something and maybe that another lost some health.
